### PR TITLE
sentinel: fix check for same reported vs spec sync standbys

### PR DESCRIPTION
**Pull request by sgotti**
When electing a new master we are swapping the new master uid with the old
master uid in the syncstandbys slice. This could end with an unordered slice in
the spec that will make the sentinel fail the check that the reported standbys
are the same of the spec one blocking any future syncstandby update.

Since the reported order is not a problem just check that the syncstandbys are
the same regardless of their order.

We'll keep the sorting to avoid unneeded updates to synchronous_standby_names by
the keeper.

### CODE REVIEW

File: pkg/util/slice_test.go [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+	}{
+		{[]string{}, []string{}, true},
+		{[]string{"", ""}, []string{""}, false},
+		{[]string{"", ""}, []string{"", ""}, true},
+		{[]string{"a", "b"}, []string{"a", "b"}, true},
+		{[]string{"a", "b"}, []string{"b", "a"}, true},
+		{[]string{"a", "b", "c"}, []string{"a", "b", "c"}, true},
+		{[]string{"a", "b", "c"}, []string{"b", "c", "a"}, true},
+		{[]string{"a", "b", "c", "a"}, []string{"a", "c", "b", "b"}, false},
+		{[]string{"a", "b", "c", "a"}, []string{"a", "c", "b", "b"}, false},
```
Comment (nh2) [inline pkg/util/slice_test.go]: May want to add some cases where the two slices have different length to get coverage of the `if len(a) != len(b) { return false }` bit (which is what makes it correct e.g. for `{abc, abcd, false}`).

**Comment by nh2** [comment]
That's great, thanks!

File: pkg/util/slice_test.go (hunk 1 above)
Comment (sgotti) [reply pkg/util/slice_test.go]: It's covered by the second test (the one with empty strings) but I'll add a more explicit test with non empty strings to make it clear.
